zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/GitNotebookRepo.java

```diff
@@ -96,9 +100,37 @@ public Revision checkpoint(String pattern, String commitMessage) {
   }
 
   @Override
-  public Note get(String noteId, Revision rev) throws IOException {
-    //TODO(bzz): something like 'git checkout rev', that will not change-the-world though
-    return super.get(noteId);
+  public synchronized Note get(String noteId, Revision rev) throws IOException {
```

Comment:
How do you think, should we have a small javaDoc comment here, explaining the algorithm of dealing with local changes? And what happens when they can not be applied without a conflict to the revision that is being requested.